vmware_dvs/agent/firewalls/vcenter_firewall.py

```diff
@@ -26,6 +31,58 @@
 CONF = config.CONF
 
 
+def firewall_updater_loop(list_queues):
+    pq = PortQueue(list_queues)
+    pq.port_updater_loop()
+    while True:
+        dvs, ports = pq.get_tasks(5)
+        if dvs and len(ports) > 0:
+            updater(dvs, ports)
+        else:
+            time.sleep(1)
+
+
+class PortQueue(object):
+    def __init__(self, list_queues):
+        self.list_queues = list_queues
+        self.taskstore = {}
+        self.networking_map = dvs_util.create_network_map_from_config(
+            CONF.ML2_VMWARE)
+
+    # Todo: add roundrobin for active DVS. SlOPS
+    def get_tasks(self, number=0):
+        for dvs in self.taskstore:
+            tasks = self.taskstore.get(dvs, [])
+            if len(tasks) > 0:
```

Comment:
lines 54-56 is equal with:
for dvs, tasks in self.taskstore.iteritems():
    if tasks: